src/scripts/slack/message.go

```diff
@@ -0,0 +1,37 @@
+package slack
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/CircleCI-Public/slack-orb-go/src/scripts/httputils"
+	"github.com/CircleCI-Public/slack-orb-go/src/scripts/jsonutils"
+)
+
+type Message struct {
+	Template     string
+	AccessToken  string
+	IgnoreErrors bool
+	Channels     []string
+}
+
+func (s *Message) PostMessage(channel string) {
```

Comment:
I've had a think about this a bit more, and I think it makes more sense from a testability perspective to return a result struct or a result, error tuple here rather than purely relying on the stdout side effect for validation.

Changing the `SendHTTPRequest` utility function to return either an actual http response, or a subset would let you surface success / failure as well as additional error messages as appropriate. Doing that would allow you to do the stdout printing from `main.go`. Just an opinion here, but I think being able to centralize the stdout writes makes following the output of the program a lot simpler. Happy to talk about this is more detail if this feedback isn't super clear in text.

Reply:
I don't agree that this comment has been resolved yet - the PostMessage is still directly logging failure messages to stdout and is also calling `os.Exit(1)` - I think those responsibilities are better left for `main.go` based on a result being returned from PostMessage. As it stands, there are side effects that are challenging to verify via test.